orchagent/muxorch.h

```diff
@@ -0,0 +1,246 @@
+    {
+        nbr_handler_->update(ip, alias, add);
+    }
+
+private:
+    bool stateActive();
+    bool stateInitActive();
+    bool stateStandby();
+
+    bool aclHandler(sai_object_id_t, bool = true);
+    bool nbrHandler(bool = true);
+
+    string mux_name_;
+
+    MuxState state_ = MuxState::MUX_STATE_INIT;
+    bool st_chg_in_progress_ = false;
+
+    IpPrefix srv_ip4_, srv_ip6_;
+    IpAddress peer_ip4_;
+
+    MuxOrch *mux_orch_;
+    MuxCableOrch *mux_cb_orch_;
+    MuxStateOrch *mux_state_orch_;
+
+    shared_ptr<MuxAclHandler> acl_handler_ = { nullptr };
+    unique_ptr<MuxNbrHandler> nbr_handler_;
+    state_machine_handlers state_machine_handlers_;
+};
+
+const request_description_t mux_cfg_request_description = {
+            { REQ_T_STRING },
+            {
+                { "server_ipv4", REQ_T_IP_PREFIX },
+                { "server_ipv6", REQ_T_IP_PREFIX },
+                { "address_ipv4", REQ_T_IP },
+            },
+            { }
+};
+
+struct NHTunnel
+{
+    sai_object_id_t nh_id;
+    int             ref_count;
+};
+
+typedef std::unique_ptr<MuxCable> MuxCable_T;
+typedef std::map<std::string, MuxCable_T> MuxCableTb;
+typedef std::map<IpAddress, NHTunnel> MuxTunnelNHs;
+
+class MuxCfgRequest : public Request
+{
+public:
+    MuxCfgRequest() : Request(mux_cfg_request_description, '|') { }
+};
+
+
+class MuxOrch : public Orch2, public Observer, public Subject
+{
+public:
+    MuxOrch(DBConnector *db, const std::vector<std::string> &tables, TunnelDecapOrch*, NeighOrch*);
+
+    typedef pair<string, bool (MuxOrch::*) (const Request& )> handler_pair;
+    typedef map<string, bool (MuxOrch::*) (const Request& )> handler_map;
+
+    bool isMuxExists(const std::string& portName) const
+    {
+        return mux_cable_tb_.find(portName) != std::end(mux_cable_tb_);
+    }
+
+    MuxCable* getMuxCable(const std::string& portName)
+    {
+        return mux_cable_tb_.at(portName).get();
+    }
+
+    MuxCable* findMuxCableInSubnet(IpAddress);
+    bool isNeighborActive(IpAddress nbr, string alias);
+    void update(SubjectType, void *);
+    void updateNeighbor(const NeighborUpdate&);
+
+    sai_object_id_t createNextHopTunnel(std::string tunnelKey, IpAddress& ipAddr);
+    bool removeNextHopTunnel(std::string tunnelKey, IpAddress& ipAddr);
+
+private:
+    virtual bool addOperation(const Request& request);
+    virtual bool delOperation(const Request& request);
+
+    bool handleMuxCfg(const Request&);
+    bool handlePeerSwitch(const Request&);
+
+    IpAddress mux_peer_switch_ = 0x0;
+    sai_object_id_t mux_tunnel_id_;
+
+    MuxCableTb mux_cable_tb_;
+    MuxTunnelNHs mux_tunnel_nh_;
+
+    handler_map handler_map_;
+
+    TunnelDecapOrch *decap_orch_;
+    NeighOrch *neigh_orch_;
+
+    MuxCfgRequest request_;
+};
+
+const request_description_t mux_cable_request_description = {
+            { REQ_T_STRING },
+            {
+                { "state",  REQ_T_STRING },
```

Comment:
missing `read_side` and `active_side` fields

Reply:
Done for the State entry